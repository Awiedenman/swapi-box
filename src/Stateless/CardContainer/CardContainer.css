.card-container {
  position: absolute;
  top: 200px;
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  width: 100%;
  background-color: #000000;
  overflow-x: scroll;
}
/*  test to make sure that it is giving me the data I want */ 