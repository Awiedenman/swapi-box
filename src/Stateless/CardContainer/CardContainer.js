import React from 'react';
import Card from '../Card/Card';
import './CardContainer.css'
import PropTypes from 'prop-types';

const CardContainer = ({ categoryData, addFavorite, favCardClicked }) => {
let selectedClass = favCardClicked === true ? 'card selected' : 'card'
  const renderCards = categoryData.map(( card, index) => 
    <Card 
    selectedClass={ selectedClass }
    key={ index }
    card={ card } 
    addFavorite={ addFavorite }
    favCardClicked={ favCardClicked }
    />
  )

  return(
    <div className='card-container'>
      { renderCards }
    </div>
  )
}



CardContainer.propTypes = {
  categoryData: PropTypes.array.isRequired,
  addFavorite: PropTypes.func.isRequired,
  favCardClicked: PropTypes.bool.isRequired 
}

export default CardContainer;
