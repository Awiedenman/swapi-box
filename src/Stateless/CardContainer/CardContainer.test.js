import React from 'react';
import CardContainer from './CardContainer';
import { shallow, mount } from 'enzyme';

describe('CardContainer', () => {
    const mockName = 'Luke';
    const mockCategoryData = [ {data: {
        Terrain: "Terrain: grasslands, mountains",
        Population: "Population: 2000000000",
        Climate: "Climate: temperate",
        Residents: "Residents: Leia Organa,Bail Prestor Organa,Raymus Antilles"
      },
      name:
      "Alderaan"
    } ];
    const mockAddFavoite = jest.fn();
    const mockFavCardClicked = true;
    const wrapper = shallow( 
          <CardContainer
            categoryData={ mockCategoryData }
            addFavorite={ mockAddFavoite }
            favCardClicked={ mockFavCardClicked }
          />
           )

  it('should match the snapshot if the cards are rendered on page', () => {
    expect( CardContainer ).toMatchSnapshot()
  })

})