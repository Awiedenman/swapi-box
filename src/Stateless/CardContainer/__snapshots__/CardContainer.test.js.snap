// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot if the cards are rendered on page 1`] = `[Function]`;
